middlewares/statsd.go

```diff
@@ -24,3 +30,77 @@ func InitClient(host string, port string, prefix string, globalTags []string) er
 func Client() *statsd.Client {
 	return c
 }
+
+
+var eventMap = map[string]statsd.EventAlertType{
+	"debug": statsd.Success,
+	"info": statsd.Info,
+	"warning": statsd.Warning,
+	"error": statsd.Error,
+	"fatal": statsd.Error,
+}
+var iso8601 = "2006-01-02T15:04:05.000Z0700"
+type logMsg struct {
+	Caller string `json:"caller"`
+	Level string `json:"level"`
+	Message string `json:"message"`
+	Name string `json:"name"`
+	Stacktrace string `json:"stacktrace"`
+	Timestamp string `json:"timestamp"`
+	Tags []string `json:"tags"`
+}
+type DataDogWriter struct {
+	client *statsd.Client
+}
+func (d DataDogWriter) Write(p []byte) (n int, err error) {
+	var msg logMsg
+	if err := json.Unmarshal(p, &msg); err != nil {
+		return 0, err
+	}
+	ts, err := time.Parse(iso8601, msg.Timestamp)
+	if err != nil {
+		return 0, err
+	}
+
+	tags := msg.Tags
+	if tags == nil {
+		tags = []string{}
+	}
+	if d.client != nil && d.client.Tags != nil {
+		tags = append(tags, d.client.Tags...)
+	}
+	evt := statsd.Event{
+		AlertType: eventMap[msg.Level],
+		Tags: tags,
+		Title: msg.Name,
+		Text: fmt.Sprintf("{\"message\": \"%s\", \"caller\": \"%s\", \"stack\": \"%s\"", msg.Message, msg.Caller, msg.Stacktrace),
```

Comment:
Extremely minor: This could be a function on the `logMsg` struct.

```
func (msg logMsg) Text() string {
    fmt.Sprintf("{\"message\": \"%s\", \"caller\": \"%s\", \"stack\": \"%s\"", msg.Message, msg.Caller, msg.Stacktrace),
}
```